ports/servoshell/desktop/app_state.rs

```diff
@@ -381,19 +387,11 @@ impl WebViewDelegate for RunningAppState {
 
     fn request_authentication(
         &self,
-        _webview: WebView,
+        webview: WebView,
         authentication_request: AuthenticationRequest,
     ) {
-        if self.inner().headless {
-            return;
-        }
-
-        if let (Some(username), Some(password)) = (
-            tinyfiledialogs::input_box("", "username", ""),
-            tinyfiledialogs::input_box("", "password", ""),
-        ) {
-            authentication_request.authenticate(username, password);
-        }
+        let authentication_dialog = Dialog::new_authentication_dialog(authentication_request);
+        self.add_dialog(webview, authentication_dialog);
```

Comment:
We need to make sure that when in headless mode we do not create the dialog. I'll make this change.

Reply:
The reason here is that if we are in headless mode, the dialog will be added to the list of dialogs, but never resolved, since there is no UI for the user to resolve it. This headless mode is used for WPT test which is why landing this hit test failures.